### ISSUE TITLE
Early stop if training loss is smaller than validation loss.

### ISSUE THREAD

**Issue by bobye** [dmlc/xgboost#1324]
I know there is an option that terminate the training if validation loss stop decreasing. But what if I want to add to this auto-stop option with triggering condition --- when training loss is smaller than validation loss. 

In my problem setting, I have a strong reason that training loss can not be smaller than validation loss, Otherwise I am certain the xgb is overfitting the validation loss. 

I didn't find a way to do so. Currently, I have to manually monitor the iterations, and pick the desired iteration numbers.

**Comment by pommedeterresautee** [comment]
You can predict the n first tree, so learn with many trees and predict with a part of them (using whatever predicate you want)
